app/models/pull_request.rb

```diff
@@ -6,7 +6,15 @@ class PullRequest  < ActiveRecord::Base
   belongs_to :github_user
   belongs_to :repository
 
-  has_many :reviews
+  has_many :reviews do
+    def create(attributes)
```

Comment:
I wouldn't change the behaviour of `create`. Instead the finder can use `.where(conditions).find_or_initialize` or `find_or_create`. That way you can always use `.update`